src/AppInstallerRepositoryCore/Rest/HttpClientHelper.cpp

```diff
@@ -66,7 +70,20 @@ namespace AppInstaller::Repository::Rest
             return ValidateAndExtractResponse(httpResponse);
     }
 
-    std::optional<web::json::value> HttpClientHelper::ValidateAndExtractResponse(const web::http::http_response& response)
+    web::http::client::http_client HttpClientHelper::GetClient(const utility::string_t& uri) const
+    {
+        web::http::client::http_client client{ uri };
+
+        // Add custom handlers if any.
+        if (m_requestHandlerStage)

```

Comment:
>m_requestHandlerStage [](start = 12, length = 21)

Somehow I feel this fits better as a parameter to this GetClient() method rather than a member variable of the class. Because making this a member variable, you are forcing all post/get methods to use the same handler.

Reply:
This intention here is to have default handler/s that are used by all post/get methods. It is possible that some of the methods would want to use custom handlers in future, at which point, the methods can be updated to do so using params. I will probably change the member varialbe's name to have default, to make it more clear.

---
In reply to: [612863834](https://github.com/microsoft/winget-cli/pull/861#discussion_r612863834) [](ancestors = 612863834)

Reply:
Then let's make GetClient() private

---
In reply to: [613490249](https://github.com/microsoft/winget-cli/pull/861#discussion_r613490249) [](ancestors = 613490249,612863834)